docs/docs/guides/evaluation/scorers.md

```diff
@@ -0,0 +1,318 @@
+
+Scorers are passed to a `weave.Evaluation` object during evaluation. There are two types of Scorers in weave:
+
+1. **Function-based Scorers:** Simple Python functions decorated with `@weave.op`.
+2. **Class-based Scorers:** Python classes that inherit from `weave.Scorer` for more complex evaluations.
+
+Scorers must return a dictionary and can return multiple metrics, nested metrics and non-numeric values such as texted returned from a LLM-evaluator about its reasoning.
+
+## Function-based Scorers
+These are functions, decorated with @weave.op that return a dictionary. They're great for simple evaluations:
+
+```python
+@weave.op
+def evaluate_uppercase(text: str):
+    return  {"text_is_uppercase": text.isupper()}
+
+eval = weave.Evaluations(..., scorers=[evaluate_uppercase])
+```
+
+When the evaluation is run, `evaluate_uppercase` checks if the text is all uppercase.
+
+## Class-based Scorers
+For more advanced evaluations, especially when you need to keep track of additional scorer metadata, try different prompts for your LLM-evaluators or make multiple function calls, you can use the `Scorer` class.
+
+**Requirements:**
+- Inherit from `weave.Scorer`.
+- Define a `score` method decorated with `@weave.op`.
+- The `score` method must return a dictionary.
+
+Example:
+
+
+```python
+from weave import Scorer
+
+class SummarizationScorer(Scorer):
+    model_id: str = "the LLM model to use"
+    system_prompt: str = "Evaluate whether the summary is good."
+
+    @weave.op
+    def some_complicated_preprocessing(text):
+        ...
+        return text
+
+    @weave.op
+    def llm_call(summary, text):
+        res = create(self.system_prompt, summary, text)
+        return {"summary_quality": res}
+
+    @weave.op
+    def score(output, text)
+        '''
+            output: The summary generated by an AI system
+            text: The original text being summarised
+        '''
+        text = some_complicated_preprocessing(text)
+        eval_result = call_llm(summary, text, self.prompt)
+        return {"summary_quality": eval_result}
+
+summarization_scorer = SummarizationScorer(model_id="o2")
+eval = weave.Evaluations(..., scorers=[summarization_scorer])
+```
+This class evaluates how good a summary is by comparing it to the original text.
+
+## How Scorers Work
+### Keyword Arguments
+Scorers can access both the output from your AI system and the input data.
+
+- **Output:** Include an `output` parameter in your scorer function's signature to access the AI system's output.
+
+- **Input:** Add parameters that match the names of the columns in your dataset to access input data.
+
+For example, if your dataset has a "news_article" column, you can access it in the scorer by adding a `news_article` parameter to your scorer's signature.
```

Comment:
aside: was this obvious to you?  This still trips me up from time to time

Reply:
updated